docs/spec/metrics.md

```diff
@@ -31,6 +31,21 @@ If you want to track the metrics in Prometheus, you need to apply analogically
 PROMETHEUS=true
 ```
 
+For custom implemented endpoints, you need to wrap the endpoint with pod name mutator in order to make the endpoint pod name visible for the cross-connect monitor, who is responsible for exposing the metrics to Prometheus:
+```
+podName := endpoint.CreatePodNameMutator()
+composite := endpoint.NewCompositeEndpoint(
+	...,
+	endpoint.NewCustomFuncEndpoint("podName", podName),
+)
+```
+
+If you deploy clients in a namespace different than `nsm-system`, use
+```
+CLIENT_NAMESPACE=<your_namespace>
+```
```

Comment:
That is strange, what if I want to deploy my clients in 2 different namespaces, what do I do?

Reply:
I removed it, as this is not directly related to the fix and needs more thought on how to apply for multiple clients & namespaces. That concept doesn't work with the current implementation and might need more changes.